src/Domain/CoordinateMaps/RotationTimeDep.hpp

```diff
@@ -0,0 +1,121 @@
+/*!
+ * \ingroup CoordinateMapsTimeDependentGroup
+ * \brief Time-dependent spatial rotation in two dimensions.
+ *
+ * Let \f$(R,\Phi)\f$ be the polar coordinates associated with
+ * \f$(\xi,\eta)\f$, where \f$\xi\f$ and \f$\eta\f$ are the unmapped
+ * coordiantes. Let \f$(r,\phi)\f$ be the polar coordinates associated with
+ * \f$(x,y)\f$, where \f$x\f$ and \f$y\f$ are the mapped coordinates.
+ * This map applies the spatial rotation \f$\phi = \Phi + \alpha(t)\f$.
+ *
+ * The formula for the mapping is:
+ *\f{eqnarray*}
+  x &=& \xi \cos \alpha(t) - \eta \sin \alpha(t), \\
+  y &=& \xi \sin \alpha(t) + \eta \cos \alpha(t).
+  \f}
+ *
+ * \note Currently, only a rotation in two-dimensional space is implemented
+ * here. In the future, this class should be extended to also support
+ * three-dimensional rotations using quaternions.
+ */
+template <>
+class RotationTimeDep<2> {
+ public:
+  static constexpr size_t dim = 2;
+
+  /// Constructor.
+  ///
+  /// \param function_of_time_name is the name of a scalar `FunctionOfTime`
+  /// that holds the rotation angle and its time derivatives
+  explicit RotationTimeDep(std::string function_of_time_name);
+  RotationTimeDep() = default;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 2> operator()(
+      const std::array<T, 2>& source_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>& map_list)
+      const noexcept;
+
+  boost::optional<std::array<double, 2>> inverse(
+      const std::array<double, 2>& target_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>& map_list)
+      const noexcept;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 2> frame_velocity(
+      const std::array<T, 2>& source_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>& map_list)
+      const noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 2, Frame::NoFrame> jacobian(
+      const std::array<T, 2>& source_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>& map_list)
+      const noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 2, Frame::NoFrame> inv_jacobian(
+      const std::array<T, 2>& source_coords, double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>& map_list)
+      const noexcept;
+
+  void pup(PUP::er& p);  // NOLINT
+
+  bool is_identity() const noexcept { return false; }
+
+ private:
+  friend bool operator==(const RotationTimeDep<2>& lhs,
+                         const RotationTimeDep<2>& rhs) noexcept;
+  std::string f_of_t_name_ = "rotation_angle";
```

Comment:
no default value .  (we realize you put it in based on Translation, but that was there before we allowed the name to be specified)